src/Sentry/SentryStructuredLogger.cs

```diff
@@ -0,0 +1,212 @@
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Debug"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogDebug(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Debug, template, parameters, configureLog);
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Info"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogInfo(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Info, template, parameters, configureLog);
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Warning"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogWarning(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Warning, template, parameters, configureLog);
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Error"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogError(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Error, template, parameters, configureLog);
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Fatal"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogFatal(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
+        {
+            CaptureLog(SentryLogLevel.Fatal, template, parameters, configureLog);
+        }
+    }
+
+    private void CaptureLog(SentryLogLevel level, string template, object[]? parameters, Action<SentryLog>? configureLog)
+    {
+        Debug.Assert(_options is not null);
```

Comment:
why not IsEnabled check here? and by pass all the public method checks

Reply:
I wanted to avoid the `CaptureLog` method invocation if logging is not enabled.
This assert is to let null-flow analysis know that options are not null here, because we already checked in the calling method.

Reply:
made `SentryStructuredLogger` abstract, and moved the `IsEnabled`-check to one of the implementation's `CaptureLog` override ... so there now is only a single `IsEnabled`-check